Error while linking with float on OS X

I'm getting linking error:
> clang-6.0: error: no such file or directory: '/usr/local/lib/R/3.5/site-library/float/libs/float.dylib'
make: *** [rsparse.so] Error 1

I believe this is because of wrong negation here:

https://github.com/wrathematics/float/blob/721e73d7ace0cf6b6a4fb9f28ce9170254417ba2/R/02-flags.r#L38

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,6 @@
 Release 0.2-2:
   * Added sign().
+  * Fixed bug with linking on OS X.
 
 Release 0.2-1:
   * Better linking/flags.

--- a/R/02-flags.r
+++ b/R/02-flags.r
@@ -6,7 +6,7 @@ get_os <- function()
 {
   os = tolower(Sys.info()["sysname"])
   names(os) = NULL
-  
+
   os
 }
 
@@ -15,18 +15,18 @@ get_os <- function()
 ldflags_string = function(static=FALSE)
 {
   install_path = "libs"
-  
+
   os = get_os()
   dynamic_link = (os == "linux" || os == "freebsd" || os == "darwin")  &&  !isTRUE(static)
-  
+
   if (nchar(.Platform$r_arch) > 0)
     path = file.path(install_path, .Platform$r_arch)
   else
     path = install_path
-  
+
   float_libs_dir_rel = system.file(path, package="float")
   float_libs_dir = tools::file_path_as_absolute(float_libs_dir_rel)
-  
+
   if (dynamic_link)
   {
     if (os == "darwin")
@@ -35,7 +35,7 @@ ldflags_string = function(static=FALSE)
       libfile = NULL
       for (libfile_test in libfiles)
       {
-        if (!file.exists(paste0(float_libs_dir, "/", libfile_test)))
+        if (file.exists(paste0(float_libs_dir, "/", libfile_test)))
         {
           libfile = libfile_test
           break
@@ -43,7 +43,7 @@ ldflags_string = function(static=FALSE)
       }
       if (is.null(libfile))
         stop(paste("unable to dynamically link: can't find any of:", paste(libfiles, collapse=", ")))
-      
+
       flags = paste0("-L", float_libs_dir, " ", float_libs_dir, "/", libfile, " -Wl,-rpath ", float_libs_dir)
     }
     else
@@ -51,14 +51,14 @@ ldflags_string = function(static=FALSE)
   }
   else
     flags = paste0(float_libs_dir, "/libfloat.a")
-  
+
   flags
 }
 
 ldflags = function(static=FALSE)
 {
   flags = ldflags_string(static=static)
-  
+
   cat(flags)
   invisible()
 }
@@ -68,15 +68,15 @@ ldflags = function(static=FALSE)
 cppflags_string = function()
 {
   install_path = "include"
-  
+
   if (nchar(.Platform$r_arch) > 0)
     path = file.path(install_path, .Platform$r_arch)
   else
     path = install_path
-  
+
   float_include_dir_rel = system.file(path, package="float")
   float_include_dir = tools::file_path_as_absolute(float_include_dir_rel)
-  
+
   flags = paste0("-I", float_include_dir)
   flags
 }
@@ -84,7 +84,7 @@ cppflags_string = function()
 cppflags = function()
 {
   flags = cppflags_string()
-  
+
   cat(flags)
   invisible()
 }

